lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -2640,6 +2642,18 @@ def GetMigrationStatus(self, instance):
                 query_migrate["ram"]["transferred"]
             migration_status.total_ram = query_migrate["ram"]["total"]
 
+            migration_status.postcopy_status = None
+            migration_caps = instance.hvparams[constants.HV_KVM_MIGRATION_CAPS]
+            # migration_caps is a ':' delimited string, so checking
+            # if 'postcopy-ram' is a substring also covers using
+            # x-postcopy-ram for QEMU 2.5
+            if migration_caps and "postcopy-ram" in migration_caps:
+              dirty_sync_count = query_migrate["ram"]["dirty-sync-count"]
+              if (migration_status.status == constants.HV_MIGRATION_ACTIVE
+                  and dirty_sync_count >= _POSTCOPY_SYNC_COUNT_THRESHOLD):
+                self.qmp.StartPostcopyMigration()
```

Comment:
Since this call has a side-effect, it would be good to log something along the lines of "switching migration to post-copy mode' here, so that it also appears in the node daemon logs.